internal/commands/app/create_inputs.go

```diff
@@ -91,6 +104,16 @@ func (i *createInputs) resolveDirectory(wd string) (string, error) {
 	if !fi.Mode().IsDir() {
 		return fullPath, nil
 	}
+	if i.From != "" && i.From == i.Directory {
```

Comment:
also, I'd like to get this flow/logic in a place where a user could provide `--yes` and be able to still by-pass any and all prompts, what do you think about the following flow:

```cmd
# inform the user and prompt them with a confirm
$ 01:23:45 UTC WARN  The directory './new-app' already exists, writing app contents to that destination may result in file conflicts.
$ Would you still like to write app contents to './new-app'? ('No' will prompt you to provide another destination) [y/N]

# if 'yes', then proceed as usual
# if 'no', follow-up with a prompt to enter a new destination/directory
```

pros:
  * keeps us`-y`-able and gives CI environments a route to still write contents
  * lets users still create an app into an existing directory (maybe it's got non-realm stuff, a pre-initialized git repo or something along those lines..)

cons:
  * forces a two-step input from a user should they want to correct this problem (rather than just one-step in your current impl)

what do you think?

Reply:
> i thought we settled on prompting the user for a new `Directory` name to write to only if the directory to write to already existed? this check looks like it does otherwise, though I may be missing something here?

The single `if` statement is for the last part of the check, you are describing. 
On lines 97-99 we determine if the file points to a file or directory or is non-existent.
On line 104 we determine if the existent path points to a file.
Which leaves us at the state that we are looking at an existing folder.
On line 107 we are checking that the existing folder is the same name as the app we are basing the new app off of.

Reply:
> also, I'd like to get this flow/logic in a place where a user could provide `--yes` and be able to still by-pass any and all prompts, what do you think about the following flow:
> 
> ```batchfile
> # inform the user and prompt them with a confirm
> $ 01:23:45 UTC WARN  The directory './new-app' already exists, writing app contents to that destination may result in file conflicts.
> $ Would you still like to write app contents to './new-app'? ('No' will prompt you to provide another destination) [y/N]
> 
> # if 'yes', then proceed as usual
> # if 'no', follow-up with a prompt to enter a new destination/directory
> ```
> 
> pros:
> 
> * keeps us`-y`-able and gives CI environments a route to still write contents
> * lets users still create an app into an existing directory (maybe it's got non-realm stuff, a pre-initialized git repo or something along those lines..)
> 
> cons:
> 
> * forces a two-step input from a user should they want to correct this problem (rather than just one-step in your current impl)
> 
> what do you think?

I do like the supporting of non-interactive commands over being slightly more verbose in interactive mode.